typedargs/metadata.py

```diff
@@ -273,12 +273,43 @@ def format_returnvalue(self, value):
         if not self.return_info.is_data:
             return None
 
+        def ignore_not_required_arg(func):
```

Comment:
I understand the desire to keep everything (including this wrapper function) in a single function for conciseness, but in this case I would recommend pulling this out to its own standalone private function:

```python
def _call_with_optional_arg(func, arg):
    """If func takes an argument, pass in arg, otherwise just call it and return the value."""
```

The advantages I see are:
- reduces the size of this larger function to make it easier to understand
- allows testability of the `_call_with_arg_if_needed` function since it's standalone and not a locally defined closure